packages/types/src/clerk.ts

```diff
@@ -673,6 +673,13 @@ export type SignUpRedirectOptions = RedirectOptions &
     initialValues?: SignUpInitialValues;
   };
 
+/**
+ * OrganizationIdentifier uniquely identifies an organization within an instance
+ */
+export type OrganizationIdentifier =
+  | { id: string; slug?: never } // Via id
+  | { id?: never; slug: string }; // Via slug
```

Comment:
This is my attempt at saying via typescript: "You can identify an org via a slug alone, or an ID alone, but not a slug and an ID, and not neither." Something like the excellent protobuf [oneof](https://github.com/hashicorp/waypoint/blob/b81cf848b44f57b95cb2e9766839582cc625d9c2/pkg/server/proto/server.proto#L1316-L1324).

I don't love this though - it doesn't read very clearly to me. I'm open to the argument that a plan-old object with both params as optional might make for a better api due to readibility.

Reply:
I've used this util a couple of times recently from [type-fest](https://github.com/sindresorhus/type-fest/tree/main) RequireOneOrNone - https://github.com/sindresorhus/type-fest/blob/main/source/require-one-or-none.d.ts

Reply:
I'm with @alexcarpenter. `type-fest` is solid.